tools/inference_engine/src/impl/model.cpp

```diff
@@ -254,6 +254,16 @@ class FusionVisitor : public GpuVisitor
                     mark_node_for_deletion(pn);
             }
         }
+        else if (auto elem_add = dynamic_cast<GpuElementwiseAdd*>(pn))
+        {
+            std::cout << "testing elementwise_add node for fusion...\n";
+            auto& inputs = pn->get_inputs();
+            if (inputs.size() == 2 && !pn->get_outputs().empty()) // don't fuse with the last output node
```

Comment:
can we get this condition ` !pn->get_outputs().empty()` above all, as currently this one is common for any type of fusion?